evennia/utils/opclasses.py

```diff
@@ -0,0 +1,278 @@
+        self.default_value = default
+        self.description = description
+        self.save_data = save_data
+
+        # Value Storage contains None until the Option is loaded.
+        self.value_storage = None
+
+        # And it's not loaded until it's called upon to spit out its contents.
+        self.loaded = False
+
+    def display(self, **kwargs):
+        """
+        Renders the Option's value as something pretty to look at.
+
+        Returns:
+            How the stored value should be projected to users. a raw timedelta is pretty ugly, y'know?
+        """
+        return self.value
+
+    def _load(self):
+        """
+        Takes the provided save data, validates it, and gets this Option ready to use.
+
+        Returns:
+            Boolean: Whether loading was successful.
+        """
+        if self.save_data is not None:
+            try:
+                self.value_storage = self.deserialize(self.save_data)
+                self.loaded = True
+                return True
+            except Exception as e:
+                _log.log_trace(e)
+        return False
+
+    def _save(self):
+        """
+        Exports the current value to an Attribute.
+
+        Returns:
+            None
+        """
+        self.handler.obj.attributes.add(self.key, category=self.handler.save_category, value=self.serialize())
+
+    def deserialize(self, save_data):
+        """
+        Perform sanity-checking on the save data. This isn't the same as Validators, as Validators deal with
+        user input. save data might be a timedelta or a list or some other object. isinstance() is probably
+        very useful here.
+
+        Args:
+            save_data: The data to check.
+
+        Returns:
+            Arbitrary: Whatever the Option needs to track, like a string or a datetime. Not the same as what
+                users are SHOWN.
+        """
+        return save_data
+
+    def serialize(self):
+        """
+        Serializes the save data for Attribute storage if it's something complicated.
+
+        Returns:
+            Whatever best handles the Attribute.
+        """
+        return self.value_storage
+
+    @property
+    def changed(self):
+        return self.value_storage != self.default_value
+
+    @property
+    def default(self):
+        return self.default_value
+
+    @property
+    def value(self):
+        if not self.loaded and self.save_data is not None:
+            self._load()
+        if self.loaded:
+            return self.value_storage
+        else:
+            return self.default
+
+    @value.setter
+    def value(self, value):
+        """
+        Takes user input, presumed to be a string, and changes the value if it is a valid input.
+
+        Args:
+            value:
+            account:
+
+        Returns:
+            None
+        """
+        final_value = self.validate(value)
+        self.value_storage = final_value
+        self.loaded = True
+        self._save()
+
```

Comment:
As mentioned in chat, for maximum flexibility, having additional get/setters that takes a `**kwargs` they can pass on into the validator (as well as into save/load for optional saving) would allow this to handle any option storage type.